packages/hub/src/lib/LazyBlob.ts

```diff
@@ -0,0 +1,92 @@
+import { createReadStream } from "node:fs";
+import { open } from "node:fs/promises";
+import type { FileHandle } from "node:fs/promises";
+import { Readable } from "node:stream";
+
+export class LazyBlob extends Blob {
+	static async create(path: string, start?: number, end?: number): Promise<LazyBlob> {
+		const lazyBlob = new LazyBlob(path, start, end);
+
+		await lazyBlob.init();
+
+		return lazyBlob;
+	}
+
+	private path: string;
+	private start: number | null;
+	private end: number | null;
+	private totalSize: number;
+
+	private constructor(path: string, start?: number, end?: number) {
+		super();
+
+		this.path = path;
+		this.start = start || null;
+		this.end = end || null;
+
+		this.totalSize = 0;
+	}
+
+	get size(): number {
+		if (this.start !== null) {
+			if (this.end !== null) {
+				return Math.abs(this.end - this.start);
+			}
+
+			return this.totalSize - this.start;
+		}
+
+		return this.totalSize;
+	}
+
+	get type(): string {
+		return "";
+	}
+
+	slice(start = 0, end = this.size): LazyBlob {
+		const slice = new LazyBlob(this.path, start, end);
```

Comment:
```suggestion
		const slice = new LazyBlob(this.path, this.start + start, Math.min(this.start + end, this.end));
```

We should also throw `new TypeError("Unsupported negative start/end on LazyBlob.slice")`  if `start < 0` or `end < 0`, in case we don't want to support this behavior (we don't really need to support it for now)